app/services/reservation_creator.rb

```diff
@@ -104,4 +107,19 @@ def status_q
     ActiveSupport::StringInquirer.new(@success.to_s)
   end
 
+  def purchase_original_cross_core_order
+    original_order = Order.find(@order.merge_with_order_id)
```

Comment:
Do we need to handle the case where a record is not found?

Reply:
No. We only get to that if-block when the order has `merge_with_order_id`